android/build.sh

```diff
@@ -1,85 +1,169 @@
 #!/usr/bin/env bash
-# Build Keyman Engine for Android, Keyman for Android, and FirstVoices Android app
-# Use '-clean' flag to clean build artifacts (won't do other build steps)
+#
+# Build Keyman Engine for Android, Keyman for Android, OEM FirstVoices Android app,
+# Samples: KMsample1 and KMSample2, Test - KeyboardHarness
 
-# Set sensible script defaults:
-# set -e: Terminate script if a command returns an error
-set -e
-# set -u: Terminate script if an unset variable is used
-#set -u: Not set because of $RELEASE_OEM
-# set -x: Debugging use, print each statement
-# set -x
+#set -x
+set -eu
 
 ## START STANDARD BUILD SCRIPT INCLUDE
 # adjust relative paths as necessary
 THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
 . "${THIS_SCRIPT%/*}/../resources/build/build-utils.sh"
 ## END STANDARD BUILD SCRIPT INCLUDE
 
+. "$KEYMAN_ROOT/resources/shellHelperFunctions.sh"
+
+# This script runs from its own folder
+cd "$THIS_SCRIPT_PATH"
+
+################################ Main script ################################
+
+builder_describe \
+  "Build Keyman Engine for Android, Keyman for Android, and FirstVoices Android app." \
+  clean \
+  build \
+  test \
+  "publish                                  Publishes the APKs to the Play Store." \
+  ":engine=KMEA                             Keyman Engine for Android" \
+  ":app=KMAPro                              Keyman for Android" \
+  ":sample1=Samples/KMSample1               Sample apps: KMSample1" \
+  ":sample2=Samples/KMSample2               Sample app:  KMSample2" \
+  ":keyboardharness=Tests/KeyboardHarness   Test app: KeyboardHarness" \
+  ":fv=../oem/firstvoices/android           OEM FirstVoices for Android app" \
+  "--ci                                     Don't start the Gradle daemon. Use for CI" \
+  "--upload-sentry                          Uploads debug symbols, etc, to Sentry" 
+
+builder_parse "$@"
+
+CONFIG="release"
+DEBUG_FLAG=""
+CI_FLAG=""
+SENTRY_FLAG=""
+
+if builder_has_option --ci; then
+  CI_FLAG="--ci"
+fi
+
+if builder_has_option --debug; then
+  CONFIG="debug"
+  DEBUG_FLAG="--debug"
+fi
+
+builder_describe_outputs \
```

Comment:
None of the rest of this file should be necessary in the same way. However, there is a limitation with `--ci` at present that I need to think about; it won't be passed to child scripts yet, but this should be straightforward enough to implement.

Instead you need to:

```
builder_run_child_actions clean configure build test publish
```

Reply:
> However, there is a limitation with --ci at present that I need to think about; it won't be passed to child scripts yet,

Also the sentry flag `--upload-sentry`